File: Recursion/incDec.cpp
#include<iostream>
using namespace std;
void incDec(int n)
{
    if(n == 0)
    {
        return;
    }
    cout << n << endl;
    incDec(n - 1);
    cout << n << endl;
}

int main()
{
    int n;
    cin >> n;
    incDec(n);
    return 0;
}

File: Recursion/incDec.java
import java.util.Scanner;
public class incDec
{
    public static Scanner scn = new Scanner(System.in);
    public static void incDec(int n)
    {
        if(n == 0)
        {
            return;
        }
        System.out.println(n);
        incDec(n - 1);
        System.out.println(n);
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        incDec(n);
    }
}